MSAL/src/public/MSALError.h

```diff
@@ -94,10 +94,11 @@ typedef NS_ENUM(NSInteger, MSALErrorCode)
         For more details check MSALOAuthErrorKey and MSALOAuthErrorDescriptionKey
         in the userInfo dictionary.
      */
-    MSALErrorInteractionRequired    = -42100,
-    MSALErrorMismatchedUser = -42101,
-    MSALErrorNoAuthorizationResponse = -42102,
-    MSALErrorBadAuthorizationResponse = -42103,
+    MSALErrorInteractionRequired        = -42100,
+    MSALErrorMismatchedUser             = -42101,
+    MSALErrorNoAuthorizationResponse    = -42102,
+    MSALErrorBadAuthorizationResponse   = -42103,
+    MSALErrorNullUser                   = -42104,
```

Comment:
I can update our guidance once the Error Code is agreed upon, but I think is exposing an issue with our model. _InteractionRequired is a control variable disguised as an Error Code._ I can imagine we'll have many Error conditions that could result in InteractionRequired as the resolution.

 Is there any way we can abstract out InteractionRequired or stack it?